### DISCUSSION TITLE
Unable to upload audio files

### DISCUSSION THREAD

**Discussion by steveava** [theyosh/TerrariumPI#788] [General]
Hi,
Looking to make use of the Audio features In TerrariumPI. I have a USB sound card attached but cant seem to upload any audio files. I've tried browsing .wav .mp3 files in the Audio/Files page but they don't appear to upload. Not sure where they are supposed to end up on the PI to check either.

![image](https://user-images.githubusercontent.com/39738639/224538387-f095efbe-4105-4873-b589-360cc4993356.png)

The sound card is recognised and appears in the New Area/Sound card menu.

Any help would be appreciated.

Thanks.
Steve.

**Comment by theyosh** [comment]
Hi, you found a bug! :)

So I updated the code to fix this. So do a `git pull` and reload your browser (hold the shift key, or clear browser cache). Than it should work.

**Comment by steveava** [comment]
Hi,

Ok, great that seems to have fixed the upload.

Created a test audio file/playlist and tried to add "Type" audio area in my existing Enclosure based on day night. Although I can fill out all the details, hitting the save button doesn't' seem to make any changes. Area settings remains on screen.

![image](https://user-images.githubusercontent.com/39738639/224550217-14214b9f-b832-4c2e-b8bc-56e2a5995ff8.png)

**Reply by theyosh** [reply]
Do a git pull and it should work. Another bug ;(

**Comment by steveava** [comment]
By the way, The name in Audio files says "None-None" for any file that's uploaded.

![image](https://user-images.githubusercontent.com/39738639/224550673-6fba8172-198f-4d22-ba6b-264d0499ecea.png)

**Reply by theyosh** [reply]
That is normal, as there is no ID3 information in the file. So you need to add some meta data into the audio files to name values here. https://www.mp3tag.de/en/ and / or https://products.groupdocs.app/metadata/wav

**Comment by steveava** [comment]
I'll take a look at the links.

Thanks.

**Comment by steveava** [comment]
Well I have everything in place I think, but cant seem to get any sound from speakers.

The Dashboard seems to show that it's on.

![image](https://user-images.githubusercontent.com/39738639/224554407-7d8c4432-0e0b-46bf-9606-ad0f879c20ea.png)

I can get sound using the uploaded files and aplay(card 1).

![image](https://user-images.githubusercontent.com/39738639/224554528-9c12865a-15c1-4837-a123-d364d2037f96.png)

Perhaps theres is a config file to adjust. Any thoughts.

**Comment by steveava** [comment]
Just got back to trying the audio again. Thought I'd use headphones to test and even though in "Area Settings" both the USB audio device and Headphones are listed in sound cards, I can't save Headphones. It's selectable and you can save it, but if you edit the Area to check, the sound card is blank. Selecting USB seems to work and save but on audio.

Again using aplay on Headphones(card 0) with the uploaded file, audio works.

**Comment by steveava** [comment]
Typo, that was "no audio"

**Comment by theyosh** [comment]
Hi, this is strange as I have tested it with both the headphones as an USB audio card
![image](https://user-images.githubusercontent.com/8080177/225439529-d89ebb51-af5c-4502-8938-e362e8e70234.png)
And both works.

So two things I can think off.

1. Volume:
Use `alsamixer` to control the volume from CLI
![image](https://user-images.githubusercontent.com/8080177/225440447-4a09516c-a6f5-4eaf-945e-edeb6728545b.png)

2. The audio files are not correct. Can you play them through the web? At least the mp3 should work. The WAV I do not know. Or upload a working mp3 file.

**Comment by steveava** [comment]
Volume seems ok for both Headphones and USB Sound card.

![image](https://user-images.githubusercontent.com/39738639/225633919-62a7e137-f959-42c9-8d3b-3cf12e019f92.png)

Don't think aplay works with mp3's but in "Audio Files" I see both uploaded files and clicking on the Green play icon, they both play local to the browser.

![image](https://user-images.githubusercontent.com/39738639/225635595-7951c3bb-b707-4127-b20b-0e47ec372f77.png)

**Comment by steveava** [comment]
Not sure if this is significant but when creating a playlist and hovering over volume, I get "Delay: 0 Minutes"

![image](https://user-images.githubusercontent.com/39738639/225636551-b7ad91d6-78a4-4a0e-809a-2e1d1d727f3f.png)

Also when you increase volume to max and save, volume in the list shows 100

![image](https://user-images.githubusercontent.com/39738639/225637851-4df1fb62-b6a5-44ff-a834-b35123e667e0.png)

But if you go back in to edit

![image](https://user-images.githubusercontent.com/39738639/225638088-4c72a71a-23b7-4ced-962f-a45469d0c410.png)

Volume is back to zero. Other options like "Shuffle" and "Repeat" keep there their saved state.

**Comment by steveava** [comment]
Just tried Audio on a different pi with fresh install and the same USB card and I get the same problem above.

By the way both Pi's are 3B+, if that makes a difference.

 
![image](https://user-images.githubusercontent.com/39738639/226128860-abf51d98-eaaf-4776-ae70-1164ce61638a.png)

**Comment by theyosh** [comment]
Hmm, I updated the GUI to change the wrong tooltip and the audio volume is now set correctly. Do a `git pull` and refresh in the browser

But I would like to ask to test with a mp3 of a minute or more audio. Make sure you can play it on you desktop with VLC.
Than make sure the playlist volume is around 80.

Everything else looks good. So when this does not help, you can try to run in debug mode to see if there are errors: https://theyosh.github.io/TerrariumPI/faq/how-to-debug-terrariumpi/

For me it is all working like expected. Both for USB and headphones.

**Comment by steveava** [comment]
Great, looks like the last changes have worked and I'm getting sound. Playlist volume is also saved, however I noticed that volume seems to be on max and even if you change volume on the playlist it has no effect. Also notice that removing the audio area from the enclosure has no effect until you reboot.

**Comment by steveava** [comment]
Just to clarify, the volume level gets saved and you can adjust it but the sound from the speakers seems to be at max.
